### ISSUE TITLE
YOLOv4 Edge TPU model

### ISSUE THREAD

**Issue by arsenal-2004** [PINTO0309/PINTO_model_zoo#63]
Hey @PINTO0309 Thank you for the great work.
I have been working with the YOLOv4 Edge TPU model from your repository, but I have been facing issues with inference times being very high. I think it might be due to most of the computational graph being on the CPU and not on the Coral, as shown by the Netron visualization below.
![image](https://user-images.githubusercontent.com/32266223/104458290-2dcf0580-55d1-11eb-924c-38583a0b90b2.png)
So, I had 2 questions regarding this:
1. How did you get the full integer model to compile, given that there were multiple dequantize nodes?
2. Do you have any ideas as to why Exp and Log seem to cause this, since they are listed in the list of supported operations for TFLite?

**Comment by PINTO0309** [comment]
>1.How did you get the full integer model to compile, given that there were multiple dequantize nodes?
>2.Do you have any ideas as to why Exp and Log seem to cause this, since they are listed in the list of supported operations for TFLite?

We need to generate models by quoting repositories that do not have **`EXP`** nodes in the first place. EdgeTPU does not support division operations in general. If it is possible to replace all division operations with multiplication and addition, the problem goes away. However, **`EXP`** is not replaceable. EdgeTPU looks like a hardware optimized for multiplication and addition.
**https://coral.ai/docs/edgetpu/models-intro/#supported-operations**

example:
NG -> 1.0 / 100.0 (divide)
OK -> 1.0 * 0.01 (mul)

**Comment by arsenal-2004** [comment]
Oh okay. I understand. But how do they support things like Softmax, if they don't allow exponentiation>

**Comment by arsenal-2004** [comment]
Also, how were you able to compile it, since the graph wasn't fully quantized? I got an error when I ran the compiler.

**Comment by PINTO0309** [comment]
I can only see how the compiler works on the surface, and since the compiler specifications are not public, I can only predict the internal workings. I do not know how the compiler implements Softmax when it converts it internally.

**Comment by PINTO0309** [comment]
>Also, how were you able to compile it, since the graph wasn't fully quantized? I got an error when I ran the compiler.

What is the model number? There are multiple variants of YoloV4 committed, so I am confused as to which one you are referring to.

**Comment by arsenal-2004** [comment]
I was referring to [YOLOv4 (031)](https://github.com/PINTO0309/PINTO_model_zoo/tree/master/031_yolov4)

**Comment by PINTO0309** [comment]
The point is to exclude complex Post-Porcess from the end of the model.
model: https://drive.google.com/u/0/uc?export=download&confirm=EPA6&id=1f8c9a1Oy5BNBlmdL-hZSo4CG_NkTHsTN
```
$ edgetpu_compiler -s yolov4_416_full_integer_quant.tflite
```

**Comment by arsenal-2004** [comment]
I got an error. Maybe you are running an earlier version of the compiler?
```
$ edgetpu_compiler -s yolov4_416_full_integer_quant.tflite 
Edge TPU Compiler version 15.0.340273435
Invalid model: yolov4_416_full_integer_quant.tflite
Model not quantized
```

**Comment by PINTO0309** [comment]
The behavior of the compiler seems to have changed since I performed the conversion six months ago.

**Comment by arsenal-2004** [comment]
Yeah, I think so as well. Anyways, thanks for the discussion. I guess it's better to work with supported models like MobileDet SSD for Coral

**Comment by PINTO0309** [comment]
@arsenal-2004 
I highly recommend this repository. The original repository has been forked and greatly optimized.
**https://github.com/hhk7734/tensorflow-yolov4.git**
